shared/common-adapters/loading-wrapper.native.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import React, {Component} from 'react'
+import {Animated} from 'react-native'
+import Box from './box'
+import Button from './button'
+import HOCTimers from './hoc-timers'
+
+import type {TimerProps} from './hoc-timers'
+
+type Props = {
+  loading: boolean,
+  style?: Object,
+  doneLoadingComponent: React$Element<*>,
+  loadingComponent: React$Element<*>,
+  duration: number,
+  debugAnim?: boolean,
+} & TimerProps
+
+type State = {
+  opacity: any,
+  loadingActive: boolean,
+  loadingActiveTimeoutId: ?number,
+}
+
+class LoadingWrapper extends Component<void, Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      opacity: new Animated.Value(1),
+      loadingActive: this.props.loading,
+      loadingActiveTimeoutId: null,
+    }
+  }
+
+  _doAnimation () {
+    const {duration} = this.props
+    this.setState({loadingActive: true})
+    this.props.clearTimeout(this.state.loadingActiveTimeoutId)
+
+    this.state.opacity.setValue(1)
+    Animated.parallel(
+      [[this.state.opacity, 0]].map(
```

Comment:
It was to do multiple animations (fadeIn while fading out) but I decided against it.

We can make this just the timing code below